src/public/media.ts

```diff
@@ -77,3 +81,233 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
+              let file = createFile(helper.assembleAttachment, helper.mediaMimeType);
+              GlobalVars.getMediaHandler(mediaResult.error, file);
+            } else {
+              let assemble: AssembleAttachment = decodeAttachment(mediaResult.mediaChunk, helper.mediaMimeType);
+              helper.assembleAttachment.push(assemble);
+            }
+          } else {
+            GlobalVars.getMediaHandler(
+              { errorCode: ErrorCode.INTERNAL_ERROR, message: 'data receieved is null' },
+              null,
+            );
+          }
+        }
+      }
+    }
+
+    GlobalVars.handlers['getMedia' + actionName] = handleGetMediaRequest;
+  }
+}
+
+/**
+ * Input parameter supplied to the select Media API
+ */
+export interface MediaInputs {
+  /**
+   * List of media types allowed to be selected
+   */
+  mediaType: MediaType;
+
+  /**
+   * max limit of media allowed to be selected in one go, current max limit is 10 set by office lens.
+   */
+  maxMediaCount: number;
+
+  /**
+   * Additional properties for customization of select media in mobile devices
+   */
+  imageProps?: ImageProps;
+}
+
+/**
+ *  All properties in ImageProps are optional and have default values in the platform
+ */
+export interface ImageProps {
+  /**
+   * Optional; Lets the developer specify the image source, more than one can be specified.
+   * Default value is both camera and gallery
+   */
+  sources?: Source[];
+
+  /**
+   * Optional; Specify in which mode the camera will be opened.
+   * Default value is Photo
+   */
+  startMode?: Mode;
+
+  /**
+   * Optional; indicate if inking on the selected Image is allowed or not
+   * Default value is true
+   */
+  ink?: boolean;
+
+  /**
+   * Optional; indicate if user is allowed to move between front and back camera
+   * Default value is true
+   */
+  cameraSwitcher?: boolean;
+
+  /**
+   * Optional; indicate if putting text stickers on the selected Image is allowed or not
+   * Default value is true
+   */
+  textSticker?: boolean;
+
+  /**
+   * Optional; indicate if image filtering mode is enabled on the selected image
+   * Default value is false
+   */
+  enableFilter?: boolean;
+}
+
+/**
+ * The modes in which camera can be launched in select Media API
+ */
+export const enum Mode {
+  Photo = 1,
+  Document = 2,
+  Whiteboard = 3,
+  BusinessCard = 4,
+  //todo: Remove Video before PR
+  Video = 5,
+}
+
+/**
+ * Specifies the image source
+ */
+export const enum Source {
+  Camera = 1,
+  Gallery = 2,
+}
+
+/**
+ * Specifies the type of Media
+ */
+export const enum MediaType {
+  Image = 1,
+  //todo: Remove Video before PR
+  Video = 2,
+  // Both image and video
+  Gallery = 3,
+}
+
+/**
+ * Input for view images API
+ */
+export interface ImageUri {
+  value: string;
+  type: ImageUriType;
+}
+
+/**
+ * ID contains a mapping for content uri on platform's side, URL is generic
+ */
+export const enum ImageUriType {
+  ID = 1,
+  URL = 2,
+}
```

Comment:
adding URL in fileFormat wouldn't be proper because URL is not a file format. Also we will be making the File class open-ended. We need URL only in case if viewImages so keeping it separately makes more sense to me.